src/middleware/firmware_update/mod.rs

```diff
@@ -0,0 +1,65 @@
+#[cfg(feature = "rock")]
+mod rockusb_fwudate;
+#[cfg(feature = "rock")]
+use self::rockusb_fwudate::new_rockusb_transport;
+#[cfg(feature = "rpi")]
+mod rpi_fwupdate;
+#[cfg(feature = "rpi")]
+use rpi_fwupdate::new_rpi_transport;
+pub mod transport;
+use self::transport::FwUpdateTransport;
+
+use super::usbboot::FlashProgress;
+pub use super::usbboot::FlashingError;
+use futures::future::BoxFuture;
+use once_cell::sync::Lazy;
+use rusb::GlobalContext;
+use std::collections::HashMap;
+use tokio::sync::mpsc::Sender;
+
+#[allow(clippy::vec_init_then_push)]
+pub static SUPPORTED_MSD_DEVICES: Lazy<Vec<(u16, u16)>> = Lazy::new(|| {
+    let mut supported = Vec::<(u16, u16)>::new();
+    #[cfg(feature = "rpi")]
+    supported.push(rpi_fwupdate::VID_PID);
+    supported
+});
+
+pub static SUPPORTED_DEVICES: Lazy<HashMap<(u16, u16), FactoryItemCreator>> = Lazy::new(|| {
+    let mut creators = HashMap::<(u16, u16), FactoryItemCreator>::new();
+    #[cfg(feature = "rpi")]
+    creators.insert(
+        rpi_fwupdate::VID_PID,
+        Box::new(|_, logging| {
+            Box::pin(async move { new_rpi_transport(&logging).await.map(Into::into) })
```

Comment:
The double `box` stems from the fact that i decided on a static `HashMap` instead of just using a `match` statement inside of `fw_update_transport`. It makes it easier to opt out of compiling a given transport.

>  transports can be turned into structs so as not to store Boxes of pinned async blocks
Im curious on how this would look like as we need dynamic behavior somewhere? There are 2 ways that comes to mind which are very difficult to implement. 1. to load a transport in static memory in a way that you can pass around Unsized `&dyn FwUpdateTransport`. 2. is To make a Transport variant which can be stored on the stack e.g.

```rust
pub enum TransportVariant {
    Rpi(ConcreteType),
    Rockusb(ConcreteType),
}
```
which is a bit more feasible, but it means boilerplate code for every function which checks which variant is initialized